pkg/console/controllers/service/controller.go

```diff
@@ -108,23 +110,55 @@ func (c *ServiceSyncController) sync() error {
 		return nil
 	case operatorsv1.Removed:
 		klog.V(4).Infoln("console is in a removed state: deleting service")
-		return c.removeService()
+		if err = c.removeService(api.OpenshiftConsoleRedirectServiceName); err != nil {
+			return err
+		}
+		return c.removeService(c.serviceName)
 	default:
 		return fmt.Errorf("unknown state: %v", operatorConfig.Spec.ManagementState)
 	}
 
 	updatedOperatorConfig := operatorConfig.DeepCopy()
-	_, _, svcErr := resourceapply.ApplyService(c.serviceClient, c.recorder, service.DefaultService(nil))
+
+	requiredSvc := service.DefaultService(updatedOperatorConfig)
+	_, _, svcErr := resourceapply.ApplyService(c.serviceClient, c.recorder, requiredSvc)
 	status.HandleProgressingOrDegraded(updatedOperatorConfig, "ServiceSync", "FailedApply", svcErr)
+	if svcErr != nil {
+		return svcErr
+	}
+
+	redirectSvcErrReason, redirectSvcErr := c.SyncRedirectService(updatedOperatorConfig)
+	status.HandleProgressingOrDegraded(updatedOperatorConfig, "RedirectServiceSync", redirectSvcErrReason, redirectSvcErr)
+	if redirectSvcErr != nil {
+		return redirectSvcErr
+	}
+
 	status.SyncStatus(c.ctx, c.operatorConfigClient, updatedOperatorConfig)
 
 	return svcErr
 }
 
-func (c *ServiceSyncController) removeService() error {
-	klog.V(2).Info("deleting console service")
-	defer klog.V(2).Info("finished deleting console service")
-	return c.serviceClient.Services(c.targetNamespace).Delete(c.ctx, service.Stub().Name, metav1.DeleteOptions{})
+func (c *ServiceSyncController) SyncRedirectService(operatorConfcig *operatorsv1.Console) (string, error) {
+	if !routesub.IsCustomRouteSet(operatorConfcig) {
```

Comment:
Since these will always appear as:

```
RedirectServiceSync: FailedDeleteRedirectService
RedirectServiceSync: FailedRedirectServiceApply
```
You can probably simplify them to:

```
RedirectServiceSync: FailedDelete
RedirectServiceSync: FailedApply
```
I believe that is the pattern we have elsewhere, such as `OAuthClientSecretSync", "FailedApply"` and `"ServiceSync", "FailedApply"`